velox/functions/lib/window/NthValueBase.h

```diff
@@ -96,6 +64,23 @@ class NthValueFunction : public exec::WindowFunction {
     partitionOffset_ += numRows;
   }
 
+ protected:
+  void setOffsetIndex(const column_index_t offsetIndex) {
+    offsetIndex_ = offsetIndex;
+  }
+
+  void setConstantOffset(const std::optional<vector_size_t>& constantOffset) {
+    constantOffset_ = constantOffset;
+  }
+
+  void setIsConstantOffsetNull(const bool isConstantOffsetNull) {
+    isConstantOffsetNull_ = isConstantOffsetNull;
+  }
+
+  void setOffsets(const VectorPtr& offsets) {
+    offsets_ = offsets;
```

Comment:
Do we really need the child class to compute offsets_ vector and pass it along ? I think we can just initialize the offsets_ vector is setOffsetIndex and remove this setter also.

Reply:
Updated, thanks.